[Term Entry] C++ Maps: .find() (#6386)
Codecademy/docs: content/cpp/concepts/maps/terms/find/find.md

---
Title: '.find()'
Description: 'Searches for an element with a specific key in a map.'
Subjects:
  - 'Computer Science'
  - 'Game Development'
Tags:
  - 'Elements'
  - 'Map'
  - 'OOP'
  - 'Objects'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> The **`.find()`** method looks for an element with a given key in a map. It returns an iterator pointing to the element if the key is found; otherwise, it returns an iterator pointing to **`map.end()`**

Reviewer: ```suggestion
The **`.find()`** method searches for an element with a given key in a `std::map`. If found, it returns an iterator to the element; otherwise, it returns an iterator to `map.end()`.
```

## Syntax

```pseudo
mapName.find(key);
```

> The `key` is the element to be searched

Reviewer: ```suggestion
**Parameters:**

- `key`: The key to search for in the map.

**Return value:**

- If the key is found, returns an iterator pointing to the key-value pair.
- If the key is not found, returns an iterator to `map.end()`.
```

> 

Reviewer: before the ## Codebyte Example block, please add an ## Example block, followed by its output wrapped in ```shell ``` block

> ## Codebyte by example

Reviewer: ```suggestion
## Codebyte Example
```

The following codebyte example demonstrates how to use `.find()` to search for a key in a map:

````codebyte/cpp
#include <iostream>
#include <map>

int main() {
  // Initializing map with items
  std::map<std::string, int> lifeSpan = {
    {"Giraffe", 26},
    {"Goat", 15},
    {"Lion", 10},
    {"Tiger", 8}
  };

  // Searching for a key
  std::string key = "Lion";
  auto it = lifeSpan.find(key);

  // Checking if the key was found
  if (it != lifeSpan.end()) {
    std::cout << key << " found! Lifespan: " << it->second << " years.\n";
  } else {
    std::cout << key << " not found in the map.\n";
  }

  // Searching for a non-existent key
  key = "Elephant";
  it = lifeSpan.find(key);

  if (it != lifeSpan.end()) {
    std::cout << key << " found! Lifespan: " << it->second << " years.\n";
  } else {
    std::cout << key << " not found in the map.\n";
  }

> }```

Reviewer: ```suggestion
}
```
